radixdlt-core/radixdlt/src/main/java/com/radixdlt/stateir/OlympiaStateIR.java

```diff
@@ -0,0 +1,108 @@
+ * losses of any kind, in tort, contract or otherwise (including but not limited to loss
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.stateir;
+
+import com.google.common.collect.ImmutableList;
+import com.radixdlt.crypto.ECPublicKey;
+import com.radixdlt.identifiers.REAddr;
+import com.radixdlt.utils.UInt256;
+import java.util.Optional;
+
+/** The intermediate representation (IR) of the Olympia ledger state. */
+public record OlympiaStateIR(
+    ImmutableList<Validator> validators,
+    ImmutableList<Resource> resources,
+    ImmutableList<Account> accounts,
+    ImmutableList<AccountBalance> balances,
+    ImmutableList<Stake> stakes) {
+
+  public record Validator(
+      ECPublicKey validatorKey,
+      String name,
+      String url,
+      boolean allowsDelegation,
+      boolean isRegistered,
+      UInt256 totalStake,
+      UInt256 totalOwnership,
+      int rakePercentage,
+      REAddr ownerAddr) {}
+
+  public record Resource(
+      REAddr addr,
+      UInt256 granularity,
+      boolean isMutable,
+      Optional<ECPublicKey> owner,
+      String symbol,
+      String name,
+      String description,
+      String iconUrl,
+      String url) {}
+
+  public record Account(REAddr addr) {}
```

Comment:
What are your thoughts on instead extracting the compressed ECPublicKey from the address? This will make it simpler on the Babylon side 👍

Reply:
You should be able to do `addr.publicKey()` and unwrap it (else raise an exception)

Reply:
sounds good